router/middleware.go

```diff
@@ -147,6 +147,12 @@ func BugSnag() Middleware {
 	})
 }
 
+type tracingMiddleware struct {
+	patterns map[http.Handler]string
```

Comment:
this won't work (I was trying to bend go here). Can't index on a function

Reply:
i think if you make it an `unsafe.Pointer` it would work, but maybe there is a better way